components/template-manager/org.wso2.carbon.event.template.manager.core/src/main/java/org/wso2/carbon/event/template/manager/core/internal/util/TemplateManagerHelper.java

```diff
@@ -238,24 +252,121 @@ public static void deployArtifacts(ScenarioConfiguration configuration,
         }
     }
 
+    /**
+     * Create a JavaScript engine packed with given scripts. If two scripts have methods with same name,
+     * later method will override the previous method.
+     *
+     * @param domain
+     * @return JavaScript engine
+     * @throws TemplateDeploymentException if there are any errors in JavaScript evaluation
+     */
+    public static ScriptEngine createJavaScriptEngine(Domain domain) throws TemplateDeploymentException {
+
+        ScriptEngineManager scriptEngineManager = new ScriptEngineManager();
+        ScriptEngine scriptEngine = scriptEngineManager.getEngineByName(TemplateManagerConstants.JAVASCRIPT_ENGINE);
```

Comment:
I have sent a new PR fixing this problem. If possible please have a look on it:
https://github.com/wso2/carbon-analytics-common/pull/288
Fix:
1. Instead of requesting Nashorn, just the name "JavaScript" is used. Java 7 returns Rhino and Java 8 return Nashorn engines (Behaviour has been tested)
2. If neither of them is available (technically engine is null), an exception will be thrown only if the templates are using this feature.
Thanks.